minorminer/layout/utils/layout_utils.py

```diff
@@ -0,0 +1,177 @@
+import math
+import random
+from collections import defaultdict
+
+import networkx as nx
+import numpy as np
+from minorminer.layout.layout import Layout
+from scipy import ndimage, spatial
+
+
+def to_vector(length, d):
+    """
+    If length is an integer, it creates a d-dimensional array with values length. Otherwise it creates an array based
+    on the length iterable and checks that is the same dimension as d. 
+    """
+    if isinstance(length, (int, float)):
+        return np.array(d*(length,))
+
+    vector = np.array(length)
+    assert vector.size == d, (
+        "You inputted a vector of size {} for a {}-dimensional space; these should match.".format(
+            vector.size, d)
+    )
+    return vector
+
+
+def lattice_points_to_length(lattice_points):
+    """
+    Converts between a number of desired lattice points in a dimension to the length in that dimension; i.e., if you 
+    want 5 lattice points in a dimension, it should be length 4 resulting in integers [0, 1, 2, 3, 4].
+    """
+    if isinstance(lattice_points, int):
+        return lattice_points - 1
+    else:
+        return tuple(x-1 for x in lattice_points)
+
+
+def convert_to_chains(placement):
```

Comment:
This function is problematic... graphs can have frozensets as nodes!  Chains can be tuples or other iterables!  This is firmly in the "asking the impossible" territory and I'd like you to refactor things to avoid its use entirely.